modules/rank.js

```diff
@@ -0,0 +1,164 @@
+const Augur = require("augurbot"),
+  Rank = require("../utils/RankInfo"),
+  config = require("../config/config.json"),
+  u = require("../utils/utils");
+
+const active = new Set();
+
+const Module = new Augur.Module()
+.addInteractionCommand({
+  name: "rank",
+  guildId: config.ldsg,
+  commandId: "894614354987733052",
+  process: async (interaction) => {
+    try {
+      const subcommand = interaction.options.getSubcommand(true);
+
+      if (subcommand === "view") {
+        // View member rankings
+        await interaction.deferReply();
+        const members = interaction.guild.members.cache;
+        const member = interaction.options.getMember("user") ?? interaction.member;
+        const leaderboard = await Module.db.user.getLeaderboard({ members, member });
+        const record = leaderboard.find(r => r.discordId == member.id);
```

Comment:
This seems inefficient to get the leaderboard just for the purpose of finding rank, when rank-finding is only one of two things performed in `Module.db.user.getLeaderboard()`. What do you think about separating `getLeaderboard` into two functions - `getLeaderboard` and `getRank`? `getLeaderboard` could call `getRank` still so that its fingerprint is unchanged for other purposes.

Reply:
Agreed that it's not particularly efficient. I initially considered a second method for `.getRank()`, but didn't want to double its code. (I hadn't considered having `.getLeaderboard()` call it.) I'll separate them out.